src/main/java/ui/Ui.java

```diff
@@ -0,0 +1,85 @@
+package ui;
+
+import task.Task;
+import taskList.TaskList;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+/**
+ * The class that deals with interactions with the user. It will give responses to the user.
+ */
+public class Ui {
+    private static final String DIVIDER = "____________________________________________________________";
+    private static final String LINE_PREFIX = "|| ";
+
+    private final Scanner in;
+    private final PrintStream out;
+
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    public Ui(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+        this.out = out;
+    }
```

Comment:
Would be helpful to indicate what each overloaded constructor does through JavaDocs